### ISSUE TITLE
Need better diagnosability if NetstandardAwareAssemblyResolver throws

### ISSUE THREAD

**Issue by KirillOsenkov** [coverlet-coverage/coverlet#1196]
Our CI is using code coverage on .NET SDK 3.0.100.

When trying to update to SDK to 5.0.301 the instrumentation fails:

```
C:\Users\kirill\.nuget\packages\coverlet.msbuild\2.6.3\build\coverlet.msbuild.targets(7,5): [coverlet] Unable to instrument module: C:\.....\Release\net472\.......dll because : The type initializer for 'Coverlet.Core.Instrumentation.NetstandardAwareAssemblyResolver' threw an exception. 
```

From looking at the static constructor, it only catches the FileNotFoundException:
https://github.com/coverlet-coverage/coverlet/blob/7221afe5e432d6d9d47670d6628967f4c255e4c9/src/coverlet.core/Instrumentation/CecilAssemblyResolver.cs#L49-L64

I think the Instrumenter should be trying to unwrap the exception before reporting it:
https://github.com/coverlet-coverage/coverlet/blob/8a59bd59c1048d6b35ab0c644716c8ecf5b496e1/src/coverlet.core/Instrumentation/Instrumenter.cs#L126

**Comment by KirillOsenkov** [comment]
I'm guessing it should be calling ToString() instead of the message, and possibly report the InnerException(s) recursively. Otherwise it's really hard to diagnose what is going on.

**Comment by KirillOsenkov** [comment]
Also perhaps it should be an error, not a warning?

**Comment by KirillOsenkov** [comment]
I had to debug using VS Child Process Attach extension, and enable first-chance exceptions to see what's going on.

![image](https://user-images.githubusercontent.com/679326/125542555-d5e3e262-c365-4a4d-bdfc-36d64fbc8032.png)

Turns out it was a bug where Mono.Cecil was bringing the wrong version of System.IO.FileSystem.Primitives.dll:
https://github.com/mono/mono/issues/12692#issuecomment-459656035

I had to update coverlet.msbuild from `2.6.3` to latest `3.0.3`. 2.6.3 shipped with Cecil 0.10.1 which had the bug. The latest 3.0.3 ships with Cecil 0.11, which targets NetStandard 2.0 and doesn't have the bug.

Long story short, this would have been easier to investigate if the Instrumenter line 126 printed the entire exception instead of just the message.

I'd immediately see that it's a MissingMethodException: `Method not found: 'Void System.IO.FileStream..ctor(System.String, System.IO.FileMode, System.IO.FileAccess, System.IO.FileShare)'.` and I could search for it to find the Cecil bug.

**Comment by MarcoRossignoli** [comment]
Thx a lot for looking into this!
Next time you can also try to use some env flag to debug https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/Troubleshooting.md#enable-collector-instrumentation-debugging
